prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt

```diff
@@ -91,24 +98,27 @@ class ProcessFhirCommands : CliktCommand(
         val inputFileType = inputFile.extension.uppercase()
         when {
             // HL7 to FHIR conversion
-            inputFileType == "HL7" && outputFormat == Report.Format.FHIR.toString() ->
-                outputResult(convertToFhir(contents, actionLogger).first, actionLogger)
+            inputFileType == "HL7" && outputFormat == Report.Format.FHIR.toString() -> {
+                outputResult(
+                    handleSenderAndReceiverTransforms(convertHl7ToFhir(contents, actionLogger).first), actionLogger
+                )
+            }
 
             // FHIR to HL7 conversion
             (inputFileType == "FHIR" || inputFileType == "JSON") && outputFormat == Report.Format.HL7.toString() -> {
-                outputResult(convertToHl7(contents))
+                outputResult(convertFhirToHl7(contents))
             }
 
             // FHIR to FHIR conversion
             (inputFileType == "FHIR" || inputFileType == "JSON") && outputFormat == Report.Format.FHIR.toString() -> {
-                outputResult(convertToFhirToFhir(contents), actionLogger)
+                outputResult(convertFhirToFhir(contents), actionLogger)
             }
 
             // HL7 to FHIR to HL7 conversion
             inputFileType == "HL7" && outputFormat == Report.Format.HL7.toString() -> {
-                val (bundle, inputMessage) = convertToFhir(contents, actionLogger)
-                val output = convertToHl7(FhirTranscoder.encode(bundle))
-                outputResult(convertToHl7(FhirTranscoder.encode(bundle)))
+                val (bundle, inputMessage) = convertHl7ToFhir(contents, actionLogger)
+                val output = convertFhirToHl7(FhirTranscoder.encode(bundle))
+                outputResult(convertFhirToHl7(FhirTranscoder.encode(bundle)))
```

Comment:
```suggestion
                outputResult(output)
```

Reply:
I just noticed this and I'm not sure why it's calling `convertFhirToHl7` twice and this seems like it would be better.

Reply:
I agree with you @adegolier. Your correction is good to me.